be/src/exec/spill/options.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "column/vectorized_fwd.h"
+#include "exec/sort_exec_exprs.h"
+#include "exec/sorting/sorting.h"
+#include "exec/spill/block_manager.h"
+
+namespace starrocks::spill {
+struct SpilledChunkBuildSchema {
+    void set_schema(const ChunkPtr& chunk) { _chunk = chunk->clone_empty(0); }
```

Comment:
why use ChunkPtr as parameter? better to store Schema here and create new chunk based on Schema in new_chunk method?

Reply:
we need match columns(nullable property and type.) and  slot_id_to_index. only schema is not enough